lua/starfall/editor/sfderma.lua

```diff
@@ -210,9 +213,45 @@ PANEL = {}
 function PANEL:Init()
 end
 
+local function moveFile(fileNode, toNode, callback)
+	if (toNode:GetFileName()) then return end
+
+	if (fileNode:GetFolder()) then
+		if (fileNode == toNode) then return end
+		if (SF.CopyFolder(fileNode:GetFolder(), toNode:GetFolder())) then
+			SF.DeleteFolder(fileNode:GetFolder())
+			local successMessage = "Successfully moved " .. fileNode:GetFolder() .. " to " .. toNode:GetFolder()
+
+			SF.AddNotify(LocalPlayer(), successMessage, "GENERIC", 7, "DRIP3")
+			print(successMessage)
+		else return end
+	elseif fileNode:GetFileName() then
+		if (SF.CopyFile(fileNode:GetFileName(), toNode:GetFolder())) then
+			file.Delete(fileNode:GetFileName())
+			local successMessage = "Successfully moved " .. string.GetFileFromFilename(fileNode:GetFileName()) .. " to " .. toNode:GetFolder()
+
+			SF.AddNotify(LocalPlayer(), successMessage, "GENERIC", 7, "DRIP3")
+			print(successMessage)
+		else return end
+	end
+
+	callback()
+end
+
+local function addDragHandling(node)
+	local setParent = node.SetParent
+	
```

Comment:
```suggestion
```
Don't need this anymore, but leaving it as a self-review before bed to test tomorrow :^)